tools/cli/src/cli/lib.rs

```diff
@@ -139,3 +179,298 @@ pub(crate) fn password_prompt(prompt: &str) -> Option<String> {
+                    .validate_with(|s: &String| -> Result<(), &str> {
+                        if VALIDATE_TOTP_RE.is_match(s) {
+                            Ok(())
+                        } else {
+                            Err("The code should be a 5 or 6 digit number")
+                        }
+                    })
+                    .interact_text()
+                    .expect("Failed to interact with interactive session");
+
+                let res = match client
+                    .idm_person_identify_user(
+                        &other_id.unwrap_or_default(),
+                        IdentifyUserRequest::SubmitCode {
+                            other_totp: other_totp.parse().unwrap_or_default(),
+                        },
+                    )
+                    .await
+                {
+                    Ok(res) => res,
+                    Err(e) => {
+                        return server_error(e);
+                    }
+                };
+                match res {
+                    IdentifyUserResponse::CodeFailure => {
+                        eprintln!("{CODE_FAILURE_ERROR_MESSAGE}");
+                        return;
+                    }
+                    IdentifyUserResponse::Success => {
+                        println!(
+                            "{}'s identity has been successfully verified 🎉🎉",
+                            other_id.unwrap_or_default()
+                        );
+                        return;
+                    }
+                    IdentifyUserResponse::InvalidUserId => {
+                        eprintln!(
+                            "{} {INVALID_USER_ID_ERROR_MESSAGE}",
+                            other_id.unwrap_or_default()
+                        );
+                        return;
+                    }
+                    IdentifyUserResponse::ProvideCode { step, totp } => {
+                        run_identity_verification_no_tui(
+                            // since we have already inserted the code, we have to go to display code second,
+                            IdentifyUserState::DisplayCodeSecond {
+                                self_totp: totp,
+                                step,
+                            },
+                            client,
+                            self_id,
+                            other_id,
+                        )
+                        .await
+                    }
+
+                    _ => {
+                        eprintln!("{INVALID_STATE_ERROR_MESSAGE}");
+                        return;
+                    }
+                }
+            }
+            IdentifyUserState::DisplayCodeFirst { self_totp, step } => {
+                println!("Provide the following code when asked: {}", self_totp);
+                let seconds_left = get_ms_left_from_now(step as u128) / 1000;
+                println!("This codes expires in {seconds_left} seconds!");
```

Comment:
Don't add the exclaimation here, it adds too much panic.

Reply:
sure :)